File: PalindromePython/string_utils.py
import re

def clean_string(text):
    return re.sub(r'[^a-zA-Z0-9]', '', text).lower()

def is_palindrome(text):
    cleaned = clean_string(text)
   
    char_array = list(cleaned)
   
    reversed_array = char_array[::-1]
    return char_array == reversed_array

def are_anagrams(text1, text2):
    cleaned1 = clean_string(text1)
    cleaned2 = clean_string(text2)
   
    array1 = list(cleaned1)
    array2 = list(cleaned2)
   
    sorted_array1 = sorted(array1)
    sorted_array2 = sorted(array2)
    return sorted_array1 == sorted_array2

File: PalindromeJava/String_utils.java
import java.util.Arrays;

class StringUtils {
    
    public static String cleanString(String text) {
        // Remove all non-alphanumeric characters and convert to lowercase
        return text.replaceAll("[^a-zA-Z0-9]", "").toLowerCase();
    }
    
    public static boolean isPalindrome(String text) {
        String cleaned = cleanString(text);
        String reversed = new StringBuilder(cleaned).reverse().toString();
        return cleaned.equals(reversed);
    }
    
    public static boolean areAnagrams(String text1, String text2) {
        String cleaned1 = cleanString(text1);
        String cleaned2 = cleanString(text2);
        
        // Convert to char arrays and sort them
        char[] chars1 = cleaned1.toCharArray();
        char[] chars2 = cleaned2.toCharArray();
        Arrays.sort(chars1);
        Arrays.sort(chars2);
        
        // Compare sorted arrays
        return Arrays.equals(chars1, chars2);
    }
}